### PR TITLE
[Snyk] Security upgrade moment from 2.29.1 to 2.29.4

### PR DESCRIPTION
**Pull request by flodnv**
These are only bug fixes being brought in so we should be fine: https://github.com/moment/moment/blob/HEAD/CHANGELOG.md

$ https://github.com/Expensify/Expensify/issues/218289
$ https://github.com/Expensify/Expensify/issues/218321

----

<p>This PR was automatically created by Snyk using the credentials of a real user.</p><br /><h3>Snyk has created this PR to fix one or more vulnerable packages in the `npm` dependencies of this project.</h3>

merge advice

#### Changes included in this PR

- Changes to the following files to upgrade the vulnerable dependencies to a fixed version:
    - package.json
    - package-lock.json

##### With an upgrade:
Severity                   | Priority Score (*)                   | Issue                   | Breaking Change                   | Exploit Maturity
:-------------------------:|-------------------------|:-------------------------|:-------------------------|:-------------------------
high severity  |  **768/1000**  <br/> **Why?** Proof of Concept exploit, Recently disclosed, Has a fix available, CVSS 7.5  | Regular Expression Denial of Service (ReDoS) <br/>[SNYK-JS-MOMENT-2944238](https://snyk.io/vuln/SNYK-JS-MOMENT-2944238) |  No  | Proof of Concept 

(*) Note that the real score may have changed since the PR was raised.

Check the changes in this PR to ensure they won't cause issues with your project.

------------

**Note:** *You are seeing this because you or someone else with access to this repository has authorized Snyk to open fix PRs.*

For more information:  
🧐 [View latest project report](https://app.snyk.io/org/expensify/project/a75415c0-01a0-4906-abb8-070d86e05d58?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr)

🛠 [Adjust project settings](https://app.snyk.io/org/expensify/project/a75415c0-01a0-4906-abb8-070d86e05d58?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr/settings)

📚 [Read more about Snyk's upgrade and patch logic](https://support.snyk.io/hc/en-us/articles/360003891078-Snyk-patches-to-fix-vulnerabilities)

[//]: # (snyk:metadata:{"prId":"b00ac7b8-a113-4074-ac97-95caeea34a68","prPublicId":"b00ac7b8-a113-4074-ac97-95caeea34a68","dependencies":[{"name":"moment","from":"2.29.1","to":"2.29.4"}],"packageManager":"npm","projectPublicId":"a75415c0-01a0-4906-abb8-070d86e05d58","projectUrl":"https://app.snyk.io/org/expensify/project/a75415c0-01a0-4906-abb8-070d86e05d58?utm_source=github&utm_medium=referral&page=fix-pr","type":"user-initiated","patch":[],"vulns":["SNYK-JS-MOMENT-2944238"],"upgrade":["SNYK-JS-MOMENT-2944238"],"isBreakingChange":false,"env":"prod","prType":"fix","templateVariants":["updated-fix-title","priorityScore","merge-advice-badge-shown"],"priorityScoreList":[768]})

---

**Learn how to fix vulnerabilities with free interactive lessons:**

 🦉 [Learn about vulnerability in an interactive lesson of Snyk Learn.](https://learn.snyk.io?loc&#x3D;fix-pr)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -68,7 +68,7 @@
     "localforage": "^1.10.0",
     "lodash": "4.17.21",
     "metro-config": "^0.64.0",
-    "moment": "^2.27.0",
+    "moment": "^2.29.4",
```
Comment (flodnv) [inline package.json:71]: Given what's in `package-lock.json`, seems this was wrong...

**Comment by chiragsalian** [comment]
you've got some tests failing. Can you update with main?

**Comment by flodnv** [comment]
I have merged main, and looked at the failing GH action, and so I ran `npm run gh-actions-build` locally and I get no diff 😕 I am not sure what's going on here, any ideas @roryabraham ?

**Comment by roryabraham** [comment]
Weird ... maybe this is a hint?

image

Unmet dependency

**Comment by flodnv** [comment]
Many thanks @roryabraham ! Updated, ready for final review @chiragsalian

**Review by chiragsalian — APPROVED** [review]
LGTM

**Pull request merged**